vivarium/core/registry.py

```diff
@@ -335,130 +363,71 @@ def divide_null(state):
     Serializers work together to convert Python objects, which may be
-    collections of many different kinds of objects, into JSON-compatible
+    collections of many different kinds of objects, into BSON-compatible
     representations. Those representations can then be deserialized to
     recover the original object.
-
-    Most serializers convert an object to a string. To implement a
-    serializer of this type, you only need to implement the
-    :py:meth:`vivarium.core.registry.Serializer.serialize_to_string` and
-    :py:meth:`vivarium.core.registry.Serializer.deserialize()` methods.
+    
+    Serializers should define one or more class attributes of the type 
+    :py:class:`bson.codec_options.TypeCodec`. If a store is assigned a
+    custom serializer using the ``_serializer`` key, serialization occurs
+    instead via the :py:meth:`vivarium.core.registry.Serializer.serialize()`
+    method and will be much slower.
+    
+    Deserialization is handled by PyMongo if the included codecs have the
+    ``bson_type``attribute and ``transform_bson()`` method. If not, the 
+    :py:meth:`vivarium.core.registry.Serializer.deserialize()` 
+    method is called instead and will be much slower.
 
     Args:
         name: Name of the serializer. Defaults to the class name.
-        exclusive_types: Types that are exclusively handled by this
-            serializer. These can be used to quickly lookup which
-            serializer handles a particular type. If an object does not
-            match any of these exclusive types, the `can_serialize`
-            method will be used as a fallback.
-
-            Note that exclusive type matching is exact--inheritance is
-            not accounted for. This means that if your serializer has an
-            exclusive type of A and B inherits from A, an instance of B
-            will not match to your serializer's exclusive types.
     """
     REGEX_FOR_NAME = re.compile('[A-Za-z0-9-_]+')
     REGEX_FOR_SERIALIZED_ANY_TYPE = re.compile(
         f'!{REGEX_FOR_NAME.pattern}\\[(.*)\\]')
 
-    def __init__(self, name='', exclusive_types=tuple()):
+    def __init__(self, name=''):
         self.name = name or self.__class__.__name__
         self.regex_for_serialized = re.compile(f'!{self.name}\\[(.*)\\]')
-        self.alternate_keys = (
-            str(exclusive_type)
-            for exclusive_type in exclusive_types
-        )
-
-    def serialize_to_string(self, data):
-        """Serialize some data.
-
-        Subclasses should override this function,
-
-        Args:
-            data: Data to serialize.
-
-        Returns:
-            The serialized data.
+        self.codecs = self.get_codecs()
+        for codec in self.codecs:
+            if hasattr(codec, 'bson_type') and hasattr(codec, 'transform_bson'):
+                self.deserialize = lambda x: x
+                self.can_deserialize = lambda _: False
+    
+    def get_codecs(self):
+        """Get list of codecs in serializer.
```

Comment:
can you add a sentence here on what codec are used for.

Reply:
Addressed in 7e14b75.